### ISSUE TITLE
BUG: Message handling out of sync

### ISSUE THREAD

**Issue by Cynical-Optimist** [buildstream-migration/bst-staging#560]
[See original issue on GitLab](https://gitlab.com/BuildStream/buildstream/-/issues/560)
In GitLab by [[Gitlab user @valentindavid]](https://gitlab.com/valentindavid) on Aug 7, 2018, 10:01

## Summary

[//]: # (Summarize the bug encountered concisely)

Here is an example of error I get in the console while running Buildstream 1.1.5. I got several of those.

```
Process Process-571:
Traceback (most recent call last):
  File "/usr/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/lib/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/home/valentin/.local/lib/python3.6/site-packages/BuildStream-1.1.5-py3.6.egg/buildstream/_scheduler/jobs/job.py", line 437, in _child_action
    self._queue.put(Envelope('child_data', self.child_process_data()))
  File "/home/valentin/.local/lib/python3.6/site-packages/BuildStream-1.1.5-py3.6.egg/buildstream/_scheduler/jobs/elementjob.py", line 113, in child_process_data
    cache_size = self._element._get_artifact_cache().calculate_cache_size()
  File "/home/valentin/.local/lib/python3.6/site-packages/BuildStream-1.1.5-py3.6.egg/buildstream/_artifactcache/cascache.py", line 463, in calculate_cache_size
    self.cache_size = utils._get_dir_size(self.casdir)
  File "/home/valentin/.local/lib/python3.6/site-packages/BuildStream-1.1.5-py3.6.egg/buildstream/utils.py", line 569, in _get_dir_size
    return get_size(path)
  File "/home/valentin/.local/lib/python3.6/site-packages/BuildStream-1.1.5-py3.6.egg/buildstream/utils.py", line 565, in get_size



BUG: Message handling out of sync, unable to retrieve failure message for element flatpak_image element at platform-arch-libs-debug.bst [line 1 column 0]
```

I think this happens when I run several builds in the same time on the same machine. Though I am not totally sure.

## Other relevant information

[//]: # (Add the BuildStream version. Substitute below the "x" by "1", "2" or the right version. Add more than one version if necessary)

* BuildStream version affected: /milestone %BuildStream_v1.2

----

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @valentindavid]](https://gitlab.com/valentindavid) on Aug 7, 2018, 10:02

It also seems that the bst hangs after.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Aug 7, 2018, 10:08

Is this a duplicate of #520 ?

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @valentindavid]](https://gitlab.com/valentindavid) on Aug 7, 2018, 10:15

It could be.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @valentindavid]](https://gitlab.com/valentindavid) on Aug 7, 2018, 10:18

Yes I get same stack trace. I get it systematically when building two different branches of freedesktop-sdk in parallel.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @valentindavid]](https://gitlab.com/valentindavid) on Aug 7, 2018, 10:18

closed

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @valentindavid]](https://gitlab.com/valentindavid) on Aug 7, 2018, 10:18

marked this issue as related to #520

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @jjardon]](https://gitlab.com/jjardon) on May 30, 2019, 15:33

reopened

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @jjardon]](https://gitlab.com/jjardon) on May 30, 2019, 15:34

closed
